### PR TITLE
NMS-14455: Install JDBC Driver on Core

### PR DESCRIPTION
**Pull request by emarsh1995**
- Edited deployment/core/getting-started.adoc
- Documented installation steps for JDBC driver on Core
- Added entry to deployment/nav.adoc

Opening as a draft PR first for feedback on paths and commands; these were copied directly from the Minion instructions for installing a JDBC driver.

### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* **If this code is likely to affect the UI, did you name your branch with `-smoke` in it to trigger smoke tests?**
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here, or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready. Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-14455

### CODE REVIEW

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+[source, karaf]
+----
+feature:list | grep mysql
+----
+
+.MySQL bundle is installed and available
+[source, output]
+----
+mysql-bundle   | 8.0.22   | x        | Started     | opennms-{page-component-version}  |
+----
```
Comment (dino2gnt) [inline docs/modules/deployment/pages/core/install-jdbc-driver.adoc:64]: This is actually a lot easier on core.

```suggestion
[[install-jdbc-driver-core]]
= Install JDBC Driver on OpenNMS {page-component-title} Core

To detect, poll, or collect any JDBC service, you must install the appropriate JDBC driver.

. Download the JDBC driver JAR file from a repository (for example, a public https://mvnrepository.com/artifact/mysql/mysql-connector-java/8.0.22[Maven repository]) or from the database vendor.
. Install the JAR file in \$OPENNMS_HOME/lib/
. Restart {page-component-title}

Testing an installed JDBC Driver
.Connect to the Karaf shell
[source,console]
----
ssh -p 8101 admin@localhost
----

.Attempt an ad-hoc collection using the database driver
[source, karaf]
----
collect org.opennms.netmgt.collectd.JdbcCollector 127.0.0.1 driver=org.mysql.jdbc.Driver collection=MySQL-Global-Stats-51 url=jdbc:mysql://OPENNMS_JDBC_HOSTNAME:3306/test
----

.If the JDBC driver is not available or the driver classname is incorrect, the collect command will return an error:
[source, karaf]
----
Collect failed with: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Unable to load driver class: java.lang.ClassNotFoundException: org.mariadb.jdbc.Driver 
----
```

**Review by dino2gnt — COMMENTED** [review]
It feels pretty cheeky to offer a multiline suggestion that replaces the entire document.  :)

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc (hunk 1 above)
Comment (emarsh1995) [reply docs/modules/deployment/pages/core/install-jdbc-driver.adoc:64]: I didn't know you could offer a full-document comment, but I love it. 😂 Thanks so much!

**Comment by dino2gnt** [comment]
There should also be a link to `core/install-jdbc-driver.adoc` from `performance-data-collection/collectors/jdbc.html` as well, similar to the mention for JDBC drivers on Minion.

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[[install-jdbc-driver-core]]
+= Install JDBC Driver on OpenNMS {page-component-title} Core
+
+To detect, poll, or collect any JDBC service, you must install the appropriate JDBC driver.
+
+. Download the JDBC driver JAR file from a repository (for example, a public https://mvnrepository.com/artifact/mysql/mysql-connector-java/8.0.22[Maven repository]) or from the database vendor.
+. Install the JAR file in \$OPENNMS_HOME/lib/
+. Restart {page-component-title}
+
+Testing an installed JDBC Driver
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/core/install-jdbc-driver.adoc:10]: This line doesn't make sense. I think it should be:
```suggestion
== Test an installed JDBC driver
```

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+.Connect to the Karaf shell
+[source,console]
+----
+ssh -p 8101 admin@localhost
+----
+
+.Attempt an ad-hoc collection using the database driver
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/core/install-jdbc-driver.adoc:17]:
```suggestion
.Attempt an ad hoc collection using the database driver
```
Inspired by our recent style guide foray in to Latin, I googled "ad hoc" and discovered that it doesn't need to be hyphenated, even when it's a modifier.

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+[source, karaf]
+----
+collect org.opennms.netmgt.collectd.JdbcCollector 127.0.0.1 driver=org.mysql.jdbc.Driver collection=MySQL-Global-Stats-51 url=jdbc:mysql://OPENNMS_JDBC_HOSTNAME:3306/test
+----
+
+.If the JDBC driver is not available or the driver classname is incorrect, the collect command will return an error:
```
Comment (Bonrob2) [inline docs/modules/deployment/pages/core/install-jdbc-driver.adoc:23]:
```suggestion
.If the JDBC driver is not available or the driver classname is incorrect, the collect command returns an error:
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Minor changes.

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc [hunk 5]
```diff
@@ -0,0 +1,27 @@
+[[install-jdbc-driver-core]]
+= Install JDBC Driver on OpenNMS {page-component-title} Core
+
+To detect, poll, or collect any JDBC service, you must install the appropriate JDBC driver.
+
+. Download the JDBC driver JAR file from a repository (for example, a public https://mvnrepository.com/artifact/mysql/mysql-connector-java/8.0.22[Maven repository]) or from the database vendor.
+. Install the JAR file in \$OPENNMS_HOME/lib/
```
Comment (emarsh1995) [inline docs/modules/deployment/pages/core/install-jdbc-driver.adoc:7]:
```suggestion
. Install the JAR file in `\$OPENNMS_HOME/lib/`.
```

File: docs/modules/deployment/pages/core/getting-started.adoc [hunk 6]
```diff
@@ -132,14 +133,14 @@ include::docker/initialize.adoc[]
...
 
-== Receive SNMP Traps/Informs
+== Receive SNMP traps/informs
 
-OpenNMS {page-component-title} core lets you receive and process SNMP Traps/Informs out of the box.
-The OpenNMS {page-component-title} core services run as an unprivileged user and can't bind on port numbers < 1024 without escalated privileges.
-For this reason, the default port for the SNMP Trap/Inform listener is set to port number 10162/udp instead of the IANA registered port number 162/udp.
-The following example shows how to configure the local firewall daemon to forward port 162/udp to 10162/udp.
+OpenNMS {page-component-title} Core lets you receive and process SNMP traps/informs out of the box.
+The OpenNMS {page-component-title} Core services run as an unprivileged user and can't bind on port numbers below 1024 without escalated privileges.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/getting-started.adoc:139]: Is "Core" an official term we are using as a proper noun, or a term used in the docs to differentiate from Minion/Sentinel.?
```suggestion
{page-component-title} Core lets you receive and process SNMP traps/informs out of the box.
{page-component-title} services run as an unprivileged user and can't bind on port numbers below 1024 without escalated privileges.
```

File: docs/modules/deployment/pages/core/getting-started.adoc [hunk 7]
```diff
@@ -158,40 +159,40 @@ include::debian-ubuntu/firewall-core.adoc[]
...
 
-You can verify your firewall and port forwarding configuration by sending an SNMP trap from a remote system to your OpenNMS {page-component-title} core instance with the following command:
+You can verify your firewall and port forwarding configuration by sending an SNMP trap from a remote system to your OpenNMS {page-component-title} Core instance:
 
 [source, console]
 ----
 snmptrap -v 2c -c public opennms-core-host '' 1.3.6.1.4.1.2021.991.17 .1.3.6.1.2.1.1.6.0 s "Milky Way"<1><2>
 ----
 <1> By default, OpenNMS uses the community string `public`.
 If you changed the community string in OpenNMS, use that name here.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/getting-started.adoc:169]:
```suggestion
If you changed the community string in {page-component-title}, use that name here.
```

File: docs/modules/deployment/pages/core/getting-started.adoc [hunk 8]
```diff
@@ -158,40 +159,40 @@ include::debian-ubuntu/firewall-core.adoc[]
...
-<2> Replace `opennms-core-host` with the IP or FQDN of your OpenNMS {page-component-title} core instance.
+<2> Replace `opennms-core-host` with the IP or FQDN of your OpenNMS {page-component-title} Core instance.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/getting-started.adoc:170]:
```suggestion
<2> Replace `opennms-core-host` with the IP or FQDN of your {page-component-title} Core instance.
```

File: docs/modules/deployment/pages/core/getting-started.adoc [hunk 9]
```diff
@@ -158,40 +159,40 @@ include::debian-ubuntu/firewall-core.adoc[]
...
 
-On RHEL/CentOS the `snmptrap` command line tool is part of the `net-snmp-utils`.
-When you run on Debian/Ubuntu, you have to install the `snmp-utils` package.
+On RHEL and CentOS, the `snmptrap` command line tool is part of the `net-snmp-utils`.
+If you run on Debian or Ubuntu, you have to install the `snmp-utils` package.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/getting-started.adoc:173]:
```suggestion
If you run on Debian or Ubuntu, the tool is part of the `snmp-utils` package.
```

File: docs/modules/deployment/pages/core/getting-started.adoc [hunk 10]
```diff
@@ -158,40 +159,40 @@ include::debian-ubuntu/firewall-core.adoc[]
 endif::[]
 ====
 
-You can verify your firewall and port forwarding configuration by sending an SNMP trap from a remote system to your OpenNMS {page-component-title} core instance with the following command:
+You can verify your firewall and port forwarding configuration by sending an SNMP trap from a remote system to your OpenNMS {page-component-title} Core instance:
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/getting-started.adoc:162]:
```suggestion
You can verify your firewall and port forwarding configuration by sending an SNMP trap from a remote system to your {page-component-title} Core instance:
```

**Review by dino2gnt — APPROVED** [review]
Looks great to me!

File: docs/modules/reference/pages/performance-data-collection/collectors/jdbc.adoc [hunk 11]
```diff
@@ -94,14 +94,14 @@ This snippet provides a sample collection definition named `opennms-stats`:
 </jdbc-collection>
 ----
 
-Once added to `etc/jdbc-datacollection-config.xml`, you can test it using the `collect` command available in the Karaf Shell:
+Once the definition is added to `etc/jdbc-datacollection-config.xml`, you can test it using the `collect` command in the Karaf Shell:
```
Comment (mmahacek) [inline docs/modules/reference/pages/performance-data-collection/collectors/jdbc.adoc:97]:
```suggestion
Once the definition is added to `$\{OPENNMS_HOME}/etc/jdbc-datacollection-config.xml`, you can test it using the `collect` command in the Karaf Shell:
```

File: docs/modules/reference/pages/performance-data-collection/collectors/jdbc.adoc [hunk 12]
```diff
@@ -37,37 +37,37 @@ Instead, you must use the service parameters to set the JDBC connection settings
...
+| JDBC password.
+|
 |===
 
 == JDBC collection configuration
 
 Understanding xref:operation:performance-data-collection/resource-types.adoc#resource-types[resource types] helps when editing collector-specific configuration files.
 
-Define JDBC collections in `etc/jdbc-datacollection-config.xml`.
+JDBC collections are defined in `etc/jdbc-datacollection-config.xml`.
```
Comment (mmahacek) [inline docs/modules/reference/pages/performance-data-collection/collectors/jdbc.adoc:70]:
```suggestion
JDBC collections are defined in `$\{OPENNMS_HOME}/etc/jdbc-datacollection-config.xml`.
```

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc [hunk 13]
```diff
@@ -0,0 +1,28 @@
+[[install-jdbc-driver-core]]
+= Install JDBC Driver on OpenNMS {page-component-title} Core
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/install-jdbc-driver.adoc:2]:
```suggestion
= Install JDBC Driver on {page-component-title} Core
```

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+
+To detect, poll, or collect any JDBC service, you must install the appropriate JDBC driver.
+
+. Download the JDBC driver JAR file from a repository (for example, a public https://mvnrepository.com/artifact/mysql/mysql-connector-java/8.0.22[Maven repository]) or from the database vendor.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/install-jdbc-driver.adoc:6]:
```suggestion
. Download the JDBC driver JAR file from a repository (for example, a public https://mvnrepository.com/artifact/mysql/mysql-connector-java/8.0.30[Maven repository]) or from the database vendor.
```

File: docs/modules/deployment/pages/core/install-jdbc-driver.adoc [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+. Install the JAR file in `\$OPENNMS_HOME/lib/`.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/install-jdbc-driver.adoc:7]:
```suggestion
. Install the JAR file in `$\{OPENNMS_HOME}/lib/`.
```

File: docs/modules/deployment/pages/minion/installing-jdbc-driver.adoc [hunk 16]
```diff
@@ -1,16 +1,14 @@
...
+To detect, poll, or collect any JDBC service, you must install the corresponding JDBC driver.
 
 We reference the `repository directory` relative to the Minion home directory.
-Depending on your operating system, the home directory is `/usr/share/minion` for Debian/Ubuntu or `/opt/minion` for CentOS/RHEL.
-
-. Download the JDBC driver JAR from a repository, for example a public link:https://mvnrepository.com/artifact/mysql/mysql-connector-java/8.0.22[Maven Repository].
-. Install the JAR file in the repository directory following the Maven repository pattern.
+Depending on your operating system, the home directory is `/usr/share/minion` for Debian and Ubuntu, or `/opt/minion` for CentOS and RHEL.
 
+. Download the JDBC driver JAR from a repository (for example, a public https://mvnrepository.com/artifact/mysql/mysql-connector-java/8.0.22[Maven repository]).
```
Comment (mmahacek) [inline docs/modules/deployment/pages/minion/installing-jdbc-driver.adoc:10]:
```suggestion
. Download the JDBC driver JAR from a repository (for example, a public https://mvnrepository.com/artifact/mysql/mysql-connector-java/8.0.30[Maven repository]).
```

File: docs/modules/deployment/pages/minion/installing-jdbc-driver.adoc [hunk 17]
```diff
@@ -1,16 +1,14 @@
...
+. Install the JAR file in the repository directory, following the Maven repository pattern.
 +
 For MySQL 8.0.22, the path would be `repositories/default/mysql/mysql-connector-java/8.0.22/mysql-connector-java-8.0.22.jar`.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/minion/installing-jdbc-driver.adoc:13]:
```suggestion
For MySQL 8.0.30, the path would be `repositories/default/mysql/mysql-connector-java/8.0.30/mysql-connector-java-8.0.30.jar`.
```

File: docs/modules/deployment/pages/minion/installing-jdbc-driver.adoc [hunk 18]
```diff
@@ -37,20 +35,20 @@ sudo vi deploy/jdbc-mysql.xml
 </features>
 -----
 
-<1> The version should match your {page-component-title} Core version.
-<2> The bundle is named `mysql-bundle` and we set the version number to `8.0.22`.
-<3> Bundle to download version `8.0.22` with Maven from official repositories.
+. The version should match your {page-component-title} Core version.
+. The bundle is named `mysql-bundle`, and the version number is set to `8.0.22`.
+. Bundle to download version `8.0.22` with Maven from official repositories.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/minion/installing-jdbc-driver.adoc:40]:
```suggestion
. The bundle is named `mysql-bundle`, and the version number is set to `8.0.30`.
. Bundle to download version `8.0.30` with Maven from official repositories.
```

File: docs/modules/deployment/pages/minion/installing-jdbc-driver.adoc [hunk 19]
```diff
@@ -37,20 +35,20 @@ sudo vi deploy/jdbc-mysql.xml
...
 
 As soon you write the file, Karaf picks up the feature and tries to install it.
-Troubleshoot the feature installation in the `data/log/karaf.log`.
+You can troubleshoot the installation using `data/log/karaf.log`.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/minion/installing-jdbc-driver.adoc:43]:
```suggestion
You can troubleshoot the installation using `$\{MINION_HOME}/data/log/karaf.log`.
```

**Review by Bonrob2 — DISMISSED** [review]
LGTM! Thanks!

File: docs/modules/deployment/pages/core/getting-started.adoc (hunk 6 above)
Comment (Bonrob2) [reply docs/modules/deployment/pages/core/getting-started.adoc:139]: That's a good point. There is no official "Core" term, it's just to differentiate between Minion/Sentinel and the core ... it should be lower case.

Comment (emarsh1995) [reply docs/modules/deployment/pages/core/getting-started.adoc:139]: Good to know - I'll go through and correct it!

File: docs/modules/deployment/pages/core/getting-started.adoc [hunk 20]
```diff
@@ -159,15 +159,15 @@ include::debian-ubuntu/firewall-core.adoc[]
...
 ====
 
-You can verify your firewall and port forwarding configuration by sending an SNMP trap from a remote system to your OpenNMS {page-component-title} Core instance:
+You can verify your firewall and port forwarding configuration by sending an SNMP trap from a remote system to your {page-component-title} core instance:
 
 [source, console]
 ----
 snmptrap -v 2c -c public opennms-core-host '' 1.3.6.1.4.1.2021.991.17 .1.3.6.1.2.1.1.6.0 s "Milky Way"<1><2>
 ----
 <1> By default, OpenNMS uses the community string `public`.
```
Comment (mmahacek) [inline docs/modules/deployment/pages/core/getting-started.adoc]:
```suggestion
<1> By default, {page-component-title} uses the community string `public`.
```

**Review by mmahacek — APPROVED** [review]
LGTM

**Pull request merged**